spec/unit/ssl/host_spec.rb

```diff
@@ -757,39 +757,51 @@ def base_json_comparison(result, json_hash)
       host.ssl_store(OpenSSL::X509::PURPOSE_SSL_SERVER)
     end
 
-    context "and the CRL needs to be retrieved" do
+    context "and the CRL is on disk" do
       before do
         @pki = PuppetSpec::SSL.create_chained_pki
 
         @revoked_cert = @pki[:revoked_root_node_cert]
 
         localcacert = Puppet.settings[:localcacert]
         Puppet::Util.replace_file(localcacert, 0644) {|f| f.write @pki[:ca_bundle] }
+        Puppet::Util.replace_file(Puppet.settings[:hostcrl], 0644) do |f|
+          f.write @pki[:crl_chain]
+        end
       end
 
       after do
         Puppet::FileSystem.unlink(Puppet.settings[:localcacert])
         Puppet::FileSystem.unlink(Puppet.settings[:hostcrl])
       end
 
-      it "a second invocation of #ssl_store returns a store without CRL checking" do
-        Puppet::SSL::CertificateRevocationList.indirection.stubs(:find).with('ca') {|ca|
-          # Mock out downloading a CRL
-          Puppet::Util.replace_file(Puppet.settings[:hostcrl], 0644) do |f|
-            f.write @pki[:crl_chain]
-          end
+      it "retrieves it and can correctly identify revoked certs" do
+        expect(@host.ssl_store.verify(@revoked_cert)).to be false
+      end
+    end
 
-          # If we were downloading the CRL we expect to be able to get a
-          # different ssl_store for that connection, one that does not have
-          # CRL checking enabled.
-          expect(@host.ssl_store.verify(@revoked_cert)).to be true
-        }.returns(true)
+    context "and the CRL is not on disk" do
+      before do
+        @pki = PuppetSpec::SSL.create_chained_pki
```

Comment:
I think this test is testing what was being tested for the previous implementation at 780 and what we're testing at 780 is tested elsewhere. IOW, I think we can get rid of the test currently at 780.